website/blog/2020-01-10-cobalt.md

```diff
@@ -0,0 +1,894 @@
+
+Thanks to recent work by @olafurpg you can now use Metals with the Pants build
+tool. The support will is the same as in case of other build tools, the Pants
+workspace will be detected automatically and an automatic import will be
+suggested.
+
+The support is currently included inside of Metals itself in contrast to other
+build tools, which have their `bloopInstall` command implemented inside Bloop.
+It is important to note that we are still working on improvements that will help
+with large codebases that Pants is usually used for.
+
+### Improvements for Bloop integration
+
+In order to make the integration between Bloop and Metals a more smooth
+experience several new improvements are now included in this release. There are
+based on feedback from the user as well us our own experiences while working
+with Metals and should make the overall experience of importing the build and
+compiling much better.
+
+#### SemanticDB is resolved inside Bloop
+
+Metals require the SemanticDB compiler plugin for a number of features to
+function properly. Because of that it was quite difficult to add it
+automatically in all supported plugins even if those supported Bloop itself.
+Perfect examples are Seed and Fury, which use Bloop to compile code, however
+adding a plugin automatically would require changes to their core code.
+
+Currently, thanks to
+[recent changes](https://github.com/scalacenter/bloop/pull/942) Metals can send
+information about the SemanticDB version, which Bloop uses to automatically
+download and configure the SemanticDB scalac plugin. Thanks to that no changes
+in the build tool itself are needed. Particular changes in regards to build
+tools are:
+
+- Gradle - predef script code used in automatic import is now simplified a lot
+  and works with even more customized workspaces. The only thing it needs to do
+  is add the Bloop plugin and run `bloopInstall`, which can easily be done
+  automatically by the user.
+- Maven - no manual configuration is now needed. To import automatically just
+  the plugin is needed to be invoked with a setting for downloading sources.
+- Sbt - the `metals-sbt` plugin is now removed and the Bloop plugin is now added
+  via automatically generated `./project/metals.sbt` file and run when imported
+  automatically. Thanks to this solution it should be possible to reuse Metals
+  compilation in the upcoming feature for offloading compilation in Sbt.
+- Mill - logic for adding semanticDB plugin is removed from Mill.
+- Pants, Fury, Seed - no changes were needed for the semanticDB integration and
+  should work out of the box.
+
+#### Switch to to using Bloop Launcher
+
+Instead of using internal Metals logic, we now use the Bloop Launcher, which
+enables us to either connect to an existing Bsp server or start a new Bloop
+instance with the specified version if nothing is running. This simplified the
+connection to Bloop greatly and enabled to work on further improvements.
```

Comment:
> This simplified the connection to Bloop greatly and enabled to work on further improvements.

-------

This greatly simplified the connection to Bloop and enabled us to work on further improvements.

Reply:
Fixed